internal/util/url.go

```diff
@@ -80,27 +114,40 @@ func FetchResource(l *log.Logger, u url.URL) ([]byte, error) {
 
 		// check if present under oemDirPath, if so use it.
 		absPath := filepath.Join(oemDirPath, path)
-		data, err := ioutil.ReadFile(absPath)
+		f, err := os.Open(absPath)
+		var closer io.Closer = f
+
 		if os.IsNotExist(err) {
 			l.Info("oem config not found in %q, trying %q",
 				oemDirPath, oemMountPath)
 
 			// try oemMountPath, requires mounting it.
-			err := mountOEM(l)
+			err = mountOEM(l)
 			if err != nil {
 				l.Err("failed to mount oem partition: %v", err)
 				return nil, ErrFailed
 			}
 
-			absPath := filepath.Join(oemMountPath, path)
-			data, err = ioutil.ReadFile(absPath)
-			umountOEM(l)
+			absPath = filepath.Join(oemMountPath, path)
+			f, err = os.Open(absPath)
+			if err != nil {
+				l.Err("failed to read oem config: %v", err)
+				return nil, ErrFailed
+			}
+
+			closer = UnmountCloser{
+				umount: func() { umountOEM(l) },
+				Closer: f,
+			}
 		} else if err != nil {
 			l.Err("failed to read oem config: %v", err)
 			return nil, ErrFailed
 		}
 
-		return data, nil
+		return struct {
+			io.Reader
+			io.Closer
+		}{bufio.NewReader(f), closer}, nil
```

Comment:
What is the point of using bufio here? If you are only exposing a ReadCloser you aren't doing anything like pushing input back into the buffer or whatnot.